### ISSUE TITLE
repeated calculation problem

### ISSUE THREAD

**Issue by shm007g** [MichalDanielDobrzanski/DeepLearningPython35#18]
I've found that you are using `sigmoid_prime` function in `network.py` and `network2.py` to calculate backpropagation. But it is not necessary.

You had calculated activation list in feedforward pass. `sigmoid_prime` is the derivative of sigmoid function, which is equal to `activation * (1 - activation) `.

 You have no need to repeated calculate this sigmoid derivative function, which is really what old-backpropa paper suggest, just one feedforward pass, and a backward pass.

**Comment by MichalDanielDobrzanski** [comment]
Closing as this library is just a ported update of neural netwotks Python code
